react/src/components/SettingPage.tsx

```diff
@@ -0,0 +1,58 @@
+import Flex from './Flex';
+import SettingPageItem from './SettingPageItem';
+import { SettingItemProps } from './SettingPageItem';
+import { SearchOutlined } from '@ant-design/icons';
+import { List, Typography, Input } from 'antd';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+
+//interface for settings
+interface SettingPageProps {
+  settingOptions: [string, SettingItemProps[]][];
+}
+
+const SettingPage: React.FC<SettingPageProps> = ({
```

Comment:
Please rename:
- SettingPageItem => SettingItem
- SettingPage => SettingList

This is because the `Page` postfix suggests it represents a specific page, such as `/summary` and `/serving`.